### PR TITLE
Fixed a whole bunch of dep warnings

### PR DESCRIPTION
**Pull request by alexmorley**
Fixed a whole bunch of dep warnings!

### CODE REVIEW

File: src/crossval.jl [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 ## cross validation generators
 
-abstract CrossValGenerator
+abstract type CrossValGenerator end
```
Comment (ararslan) [inline src/crossval.jl]: This is 0.6-only syntax. To support earlier versions, it needs to have an `@compat` in front.

**Comment by ararslan** [comment]
Nice, thanks for the contribution.

File: src/classification.jl [hunk 2]
```diff
@@ -107,13 +107,13 @@ immutable LabelMap{K}
     vs::Vector{K}
     v2i::Dict{K,Int}
 
-    function LabelMap(vs, v2i)
+    function LabelMap{K}(vs, v2i) where K
```
Comment (ararslan) [inline src/classification.jl]: Ah, I missed this on the earlier pass. This is also 0.6-only. At this point I think we can just drop support for earlier versions of Julia and update the REQUIRE to `julia 0.6`.

Comment (alexmorley) [inline src/classification.jl]: I think that's a good idea. Shall I get rid of the other compat's then?

**Comment by ararslan** [comment]
Can you update the Travis YAML to remove 0.4 and 0.5 and put in 0.6? And remove release, if that's in there.

File: src/classification.jl (hunk 2 above)
Comment (ararslan) [inline src/classification.jl]: Sure, that'd be great, thanks. After that you should be able to drop the dependency on Compat altogether.

**Pull request merged**